starknet_py/tests/e2e/mock/compile_contracts.sh

```diff
@@ -1,32 +1,81 @@
 #!/bin/bash
-
+set -e
 MOCK_DIRECTORY=starknet_py/tests/e2e/mock
-CONTRACTS_DIRECTORY="$MOCK_DIRECTORY"/contracts
-CONTRACTS_COMPILED_DIRECTORY="$MOCK_DIRECTORY"/contracts_compiled
+CONTRACTS_DIRECTORY_V1="$MOCK_DIRECTORY"/contracts_v1
+CONTRACTS_DIRECTORY_V2="$MOCK_DIRECTORY"/contracts_v2
 
-# delete all artifacts except precompiled ones
-find $CONTRACTS_COMPILED_DIRECTORY -maxdepth 1 -type f -delete
+setup_asdf() {
+    if ! command -v asdf >/dev/null 2>&1; then
+        echo "Install asdf, before executing this script is required to manage scarb version"
+        exit 1
+    fi
+    asdf plugin-add scarb
+}
 
-# compile Cairo test contracts
-echo "Compiling Cairo contracts with $(poetry run starknet-compile-deprecated --version)"
+compile_contracts_with_scarb() {
+    CONTRACTS_DIRECTORY=$1
+    SCARB_WITH_VERSION=$(cat "$CONTRACTS_DIRECTORY/.tool-versions")
 
-number_of_contracts=0
-for contract in "$CONTRACTS_DIRECTORY"/*.cairo; do
-    basename=$(basename "$contract")
+    setup_asdf
 
-    output=$CONTRACTS_COMPILED_DIRECTORY/"${basename%.*}_compiled.json"
-    abi=$CONTRACTS_COMPILED_DIRECTORY/"${basename%.*}_abi.json"
+    asdf install $SCARB_WITH_VERSION
 
-    # set account contract flag
-    account_contract_flag=""
-    if [[ $basename == *"account"* ]]; then
-      account_contract_flag="--account_contract"
-    fi
+    echo "Compiling Cairo contracts with $SCARB_WITH_VERSION"
+
+    (cd $CONTRACTS_DIRECTORY && scarb clean && scarb build)
```

Comment:
let's not run those commands in a different sunshell, there is no need to that.
instead cd we should use pushd and popd